core/packages/PackageDB.h

```diff
@@ -49,12 +49,17 @@ class PackageDB final {
 
     const std::string_view errorHint() const;
 
+    const std::string_view getLayerName(uint8_t layerRank) const;
+    const uint8_t getLayerRank(std::string layerName) const;
+    const std::size_t getLayerCount() const;
```

Comment:
This is a nit, but I think a clearer API might be

`const std::vector<Layer> &layers() const;` (this is less true if you end up storing these as a map)

to support `Layer::show` you can make `Layer` a friend class and let it access the layer storage directly.

Reply:
`const vector<Foo>&` would not copy the vector since it's a ref and doesn't allow mutation. (You could cast your way out of that but at that point you're really fighting the system intentionally)